### ISSUE TITLE
[0.10.10][Windows] Tab size is not working properly

### ISSUE THREAD

**Issue by rebornix** [Microsoft/vscode#4000]
I'm using the latest version on my Windows machine, the tab size setting is totally broken. Even though the default setting is 4, the output of Tab on my VS Code is always 8.

For instance, I added a new line starting with two tabs

![image](https://cloud.githubusercontent.com/assets/876920/13697428/686a4f84-e7a8-11e5-9bdf-de34cadccd19.png)

VS Code takes these two tabs as 8 spaces (4 for each). However if you open the file in notepad or run `git diff` against the file, you'll see it's not the case

![image](https://cloud.githubusercontent.com/assets/876920/13697477/a1380842-e7a8-11e5-90d0-e3e0f9753d0c.png)

Lastly, if we change the tab.size to 2 characters in `setting.json`, CTRL-S the setting file. The tab changes to 2 chars immediately, But once you move your cursor to the left side document, it will revert back real quick.

![vscode-tab-broken](https://cloud.githubusercontent.com/assets/876920/13697624/97708ae0-e7a9-11e5-8d01-50a03253c2bb.gif)

The biggest disappointment is no matter what I set in `setting.json`, a tab is always 8 spaces in the final output.

**Comment by alexandrudima** [comment]
fyi @isidorn I think the problem shown in the last gif is the previous status bar implementation that would always update editor options when changing active editors (that we shipped in Feb).

@rebornix We also noticed some of these issues and reworked how this works under the hood. The indentation settings used to live on the editor and now they moved to live on the model (the buffer). We also added a new setting `editor.detectIndentation` that will scan the file contents for detecting what indentation should be used.

One thing I didn't understand from your description is
> "Even though the default setting is 4, the output of Tab on my VS Code is always 8."
